sw/blas/gemm/verify.py

```diff
@@ -10,23 +10,27 @@
 import numpy as np
 from data.datagen import golden_model
 
-sys.path.append(str(Path(__file__).parent / '../../../util/sim/'))
+sys.path.append(str(Path(__file__).parent / "../../../util/sim/"))
 import verification  # noqa: E402
 from elf import Elf  # noqa: E402
 from data_utils import from_buffer, ctype_from_precision_t  # noqa: E402
 
 
-ERR_THRESHOLD = 0.001
+ERR_THRESHOLD = {8: 1e-6, 4: 1e-6, 2: 1e-2, 1: 1e-1}
```

Comment:
Can the absolute error actually be this large when using fp16 and fp8?

Reply:
It really depends on the problem size. The larger it is, the higher is the chance that you have some kind of catastrophic cancellation or saturation.

The error threshold for fp8 might even be higher. The problem is that it is almost impossible to verify fp8 kernels. fp8 is not a standard and lacks support in numpy, and we are also missing tracing of fp8 values currently, which makes it quite hard to debug and verify.